server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -1189,6 +1189,7 @@ public void writeTo(StreamOutput out) throws IOException {
         );
         // The fields behind the feature flag should always be last.
         if (DataStream.isFailureStoreFeatureFlagEnabled()) {
+            // Should be removed after backport
```

Comment:
Maybe add the same `@UpdateForV9` annotation here?

Reply:
I wanted to, but I couldn't put it on top of a statement :( I am planning to address this immediately after this PR, so I won't forget.